src/common/options.h

```diff
@@ -13,19 +13,19 @@
 
 struct Option {
   enum type_t {
-    TYPE_UINT,
-    TYPE_INT,
-    TYPE_STR,
-    TYPE_FLOAT,
-    TYPE_BOOL,
-    TYPE_ADDR,
-    TYPE_ADDRVEC,
-    TYPE_UUID,
-    TYPE_SIZE,
-    TYPE_SECS,
+    TYPE_UINT = 0,
```

Comment:
I know this macro trick looks hacky and anti-OOP but until C++20 brings some reflection features to C++, I think this used in small doses is really useful and even less prone to copy-pasting errors:

```c++
#define TYPE_META(f) \
  /* f(LABEL, ID, TYPE_STR, CEPH_TYPE,  PY_FROM, PY_TO) */ \
  f(TYPE_UINT, 1, "uint64_t", "CephInt", PyInt_FromLong, PyInt_AsLong) \
  f(TYPE_STR, 2, "std::str", "CephString", PyString_FromString, PyString_AsString) \
  ...

typedef enum {
  #define META_TO_ENUM(a, b, ...) a = b,
  TYPE_META(META_TO_ENUM)
} type_t;

static const char *type_to_c_type_str(type_t t) {
  switch(t) {
    #define META_TO_STR(a, b, c, ...) case a: return c;
    TYPE_META(META_TO_STR)
    default: return "unknown";
  }
}
```